### PR TITLE
Fixes splitTestGroup ads should not be shown when AdvertiserSplitTestStudy is not defined in seed

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/15164

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See https://github.com/brave/brave-browser/issues/15164

### CODE REVIEW

**Review by moritzhaller — APPROVED** [review]
++

File: vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/split_test_frequency_cap.cc [hunk 1]
```diff
@@ -57,8 +57,8 @@ bool SplitTestFrequencyCap::DoesRespectCap(const CreativeAdInfo& ad) const {
   }
 
   if (ad.split_test_group.empty()) {
-    // Always respect cap if there is no split testing group in the catalog
-    return true;
+    // Do not respect cap if there is no split testing group in the catalog
```
Comment (moritzhaller) [inline vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/split_test_frequency_cap.cc:60]: we should flip the return value for the conditional in L56

**Review by moritzhaller — CHANGES_REQUESTED** [review]

File: vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/split_test_frequency_cap.cc (hunk 1 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/split_test_frequency_cap.cc]: Discussed issue on DM, pushed changes

File: vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/split_test_frequency_cap_unittest.cc [hunk 2]
```diff
@@ -51,7 +50,21 @@ TEST_F(BatAdsSplitTestFrequencyCapTest, AllowIfNoGroupInCatalog) {
...
+
+  // Act
+  SplitTestFrequencyCap frequency_cap;
+  const bool should_exclude = frequency_cap.ShouldExclude(ad);
+
+  // Assert
+  EXPECT_TRUE(should_exclude);
+}
+
+TEST_F(BatAdsSplitTestFrequencyCapTest, AllowIfFieldTrialAndNoAdGroup) {
```
Comment (moritzhaller) [inline vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/split_test_frequency_cap_unittest.cc:67]: Test logic different to test name. Should add

```
  scoped_refptr<base::FieldTrial> trial = CreateFieldTrial(kTrial);
  trial->AppendGroup(kGroup, 100);
```
in line 72 and remove line 71

**Review by moritzhaller — CHANGES_REQUESTED** [review]
see comments

File: vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/split_test_frequency_cap_unittest.cc [hunk 3]
```diff
@@ -35,13 +35,12 @@ class BatAdsSplitTestFrequencyCapTest : public UnitTestBase {
...
-TEST_F(BatAdsSplitTestFrequencyCapTest, AllowIfNoGroupInCatalog) {
+TEST_F(BatAdsSplitTestFrequencyCapTest,
+    AllowIfNoFieldTrialAndNoAdGroup) {
   // Arrange
   CreativeAdInfo ad;
   ad.creative_set_id = kCreativeSetId;
-
-  scoped_refptr<base::FieldTrial> trial = CreateFieldTrial(kTrial);
-  trial->AppendGroup(kGroup, 100);
+  ad.split_test_group = "GroupA";
```
Comment (moritzhaller) [inline vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/split_test_frequency_cap_unittest.cc:43]: remove as `*NoAdGroup`

Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/split_test_frequency_cap_unittest.cc:43]: Fixed

File: vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/split_test_frequency_cap_unittest.cc (hunk 2 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/frequency_capping/exclusion_rules/split_test_frequency_cap_unittest.cc:67]: Fixed

**Review by moritzhaller — APPROVED** [review]
LGTM

**Pull request merged**